src/Umbraco.Web.Common/RuntimeMinification/SmidgeRuntimeMinifier.cs

```diff
@@ -21,31 +21,39 @@ public class SmidgeRuntimeMinifier : IRuntimeMinifier
         private readonly IHostingEnvironment _hostingEnvironment;
         private readonly ISmidgeConfig _smidgeConfig;
         private readonly IConfigManipulator _configManipulator;
-        private readonly PreProcessPipelineFactory _preProcessPipelineFactory;
         private readonly IBundleManager _bundles;
         private readonly SmidgeHelperAccessor _smidge;
 
-        private PreProcessPipeline _jsPipeline;
-        private PreProcessPipeline _cssPipeline;
+        // used only for minifying in MinifyAsync not for an actual pipeline
+        private Lazy<PreProcessPipeline> _jsMinPipeline;
+        private Lazy<PreProcessPipeline> _cssMinPipeline;
+
+        // default pipelines for processing js/css files for the back office
+        private Lazy<PreProcessPipeline> _jsPipeline;
+        private Lazy<PreProcessPipeline> _cssPipeline;
 
         public SmidgeRuntimeMinifier(
             IBundleManager bundles,
             SmidgeHelperAccessor smidge,
-            PreProcessPipelineFactory preProcessPipelineFactory,
             IHostingEnvironment hostingEnvironment,
             ISmidgeConfig smidgeConfig,
             IConfigManipulator configManipulator)
         {
             _bundles = bundles;
             _smidge = smidge;
-            _preProcessPipelineFactory = preProcessPipelineFactory;
             _hostingEnvironment = hostingEnvironment;
             _smidgeConfig = smidgeConfig;
             _configManipulator = configManipulator;
-        }
 
-        private PreProcessPipeline JsPipeline => _jsPipeline ??= _preProcessPipelineFactory.Create(typeof(JsMinifier));
-        private PreProcessPipeline CssPipeline => _cssPipeline ??= _preProcessPipelineFactory.Create(typeof(NuglifyCss));
+            _jsMinPipeline = new Lazy<PreProcessPipeline>(() => _bundles.PipelineFactory.Create(typeof(JsMinifier)));
```

Comment:
Should this not be NugilifyJs now?

Reply:
It could be, i haven't tested that. This is purely used only for minifying Action results which IMO will be better to use JsMinifier since it's a simple minifier and very fast compared to nuglify and normally this is only used to minify small json results.